lib/scholar/metrics/classification.ex

```diff
@@ -603,58 +665,58 @@ defmodule Scholar.Metrics.Classification do
       #Nx.Tensor<
         f32[3]
         [0.6666666865348816, 0.6666666865348816, 0.4000000059604645]
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.u32(2), num_classes: 3)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, beta: Nx.u32(2), num_classes: 3)
       #Nx.Tensor<
         f32[3]
         [0.6666666865348816, 0.5555555820465088, 0.625]
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.f32(0.5), num_classes: 3)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, beta: Nx.f32(0.5), num_classes: 3)
       #Nx.Tensor<
         f32[3]
         [0.6666666865348816, 0.8333333134651184, 0.29411765933036804]
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.u32(2), num_classes: 3, average: :macro)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, beta: Nx.u32(2), num_classes: 3, average: :macro)
       #Nx.Tensor<
         f32
         0.6157407760620117
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.u32(2), num_classes: 3, average: :weighted)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, beta: Nx.u32(2), num_classes: 3, average: :weighted)
       #Nx.Tensor<
         f32
         0.5958333611488342
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.f32(0.5), num_classes: 3, average: :micro)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, beta: Nx.f32(0.5), num_classes: 3, average: :micro)
       #Nx.Tensor<
         f32
         0.6000000238418579
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), Nx.tensor(0.5), num_classes: 2, average: :none)
+      iex> Scholar.Metrics.Classification.fbeta_score(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), beta: Nx.tensor(0.5), num_classes: 2, average: :none)
       #Nx.Tensor<
         f32[2]
         [0.0, 0.0]
       >
-      iex> Scholar.Metrics.Classification.fbeta_score(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), 0.5, num_classes: 2, average: :none)
+      iex> Scholar.Metrics.Classification.fbeta_score(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), beta: 0.5, num_classes: 2, average: :none)
       #Nx.Tensor<
         f32[2]
         [0.0, 0.0]
       >
   """
-  deftransform fbeta_score(y_true, y_pred, beta, opts \\ []) do
-    fbeta_score_n(y_true, y_pred, beta, NimbleOptions.validate!(opts, @fbeta_score_schema))
+  deftransform fbeta_score(y_true, y_pred, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @fbeta_score_schema)
+    {beta, opts} = Keyword.pop(opts, :beta)
+    fbeta_score_n(y_true, y_pred, beta, opts)
   end
 
   defnp fbeta_score_n(y_true, y_pred, beta, opts) do
-    check_shape(y_pred, y_true)
-    num_classes = check_num_classes(opts[:num_classes])
-    average = opts[:average]
+    check_beta(beta)
```

Comment:
I think we can remove this. It has a default value and it cannot be nil as an option.

Reply:
I'm not defining a default value of beta in `fbeta_score_schema`, only in `precision_recall_fscore_support_schema`, which was the reason to leave the check there.

Reply:
If it is required there, why not make it an argument then?

Reply:
That was the idea, but we've moved to this instead in order to be consistent with the other functions that use the beta parameter.
https://github.com/elixir-nx/scholar/pull/186#discussion_r1352874935

Reply:
Maybe using the [NimbleOptions schema options](https://hexdocs.pm/nimble_options/NimbleOptions.html#module-schema-options) with the required as true, would be better in this case?

Reply:
Honestly, I would prefer to keep it as an argument, but it is not a strong opinion. Or we remove the function and make it an option on f1_score.

Reply:
In another kind of library I would agree, but for these kinds of data science/statistical learning libraries, I'm partial to having a consistent/predictable API.